cmd/admin-handlers.go

```diff
@@ -1558,41 +1559,170 @@ func (a adminAPIHandlers) HealthInfoHandler(w http.ResponseWriter, r *http.Reque
 	}
 	defer nsLock.Unlock(lkctx.Cancel)
 
-	go func() {
-		defer close(healthInfoCh)
+	hostAnonymizer := createHostAnonymizer()
+	// anonAddr - Anonymizes hosts in given input string.
+	anonAddr := func(addr string) string {
+		newAddr, found := hostAnonymizer[addr]
+		if found {
+			return newAddr
+		}
 
-		partialWrite(healthInfo) // Write first message with only version populated
+		if host, err := xnet.ParseHost(addr); err == nil {
+			newAddr, found := hostAnonymizer[host.Name]
+			if found {
+				return newAddr
```

Comment:
Doesn't this trim everything but the host?

Reply:
It's not yet ready for review - so I have kept it in `draft` mode and was doing self-review and adding notes for myself.. But yes, your are right. It would trim things and it is actually not required. Will remove it.